### DISCUSSION TITLE
The ai does not respond after Intranet penetration

### DISCUSSION THREAD

**Discussion by sizukamasou** [waylaidwanderer/PandoraAI#130] [Q&A]
I have successfully deployed the node-chatgpt-api and pandoraAI.

And I can run it on my computer. Talk to bing normally.

I then mapped the access address 192.168.0.101:3000 to a URL using ngrok.

My other devices can access this mapped URL, but cannot interact with the ai.

I don't know how to set it up so that I can access my deployed pandoraAI from another device or location.

Here is my configuration:
![image](https://github.com/waylaidwanderer/PandoraAI/assets/134895951/85cfa820-f3df-43ac-873f-278d025c5906)
![image](https://github.com/waylaidwanderer/PandoraAI/assets/134895951/d4ad6c45-454c-49f5-9499-f8ff25305ebc)

I'm using clash's global proxy, ports 7890

Similarly, I've deployed pandora in the past，It is chatgpt3.5.
It works fine with the same through-mapping method

Please help me, thank you!

**Comment by waylaidwanderer** [comment]
Make sure you're not running the client in dev mode if you're accessing it from outside. You have to build it and run the production build instead.

**Comment by waylaidwanderer** [comment]
Oh, and the API_BASE_URL needs to be behind Ngrok as well since the client will be connecting to this URL client-side.

**Comment by sizukamasou** [comment]
> 哦，API_BASE_URL 也需要在 Ngrok 后面，因为客户端将连接到此 URL 客户端。





> Oh, and the API_BASE_URL needs to be behind Ngrok as well since the client will be connecting to this URL client-side.

Still no success
I ran npm run buildmingl as you suggested，The first time I opened the page, it showed 500 errors.

Then chatgpt told that it was missing the configuration of setting.js under pandoraAI,Originally I couldn't find it on github, so I asked gpt to tell me what was in it, and I tried to create it

![image](https://github.com/waylaidwanderer/PandoraAI/assets/134895951/28d67972-545c-4c4f-93c4-c2ae132427a6)
(https://7v4x539662.imdo.co/It is the domain name mapped by ddns)

then

I tried this domain name in api_base_url under the.env file
![image](https://github.com/waylaidwanderer/PandoraAI/assets/134895951/c0788f94-c1a0-4391-9b6f-9483bf7c0d17)
I also tried using http://localhost:3001

And shut down the firewall on the computer
I then successfully started npm run build
![image](https://github.com/waylaidwanderer/PandoraAI/assets/134895951/4e1ba119-b192-45a3-9f5f-552e55d8d941)

I can then communicate with the ai through this domain name
![image](https://github.com/waylaidwanderer/PandoraAI/assets/134895951/7e443bfc-992f-4c4c-bb72-cc48669fd390)

But the phone still couldn't interact
![62A1F7E584C6E3985C62308F6386BA23](https://github.com/waylaidwanderer/PandoraAI/assets/134895951/8b7ca2e3-c146-49d0-a770-a446fc9eba66)

I really don't know how to change my configuration,Or is there a problem with my ddns software?

**Comment by waylaidwanderer** [comment]
This isn't a bug with the repo so I'm converting this issue to a Help thread.

**Comment by minatoAI** [comment]
![image](https://github.com/waylaidwanderer/PandoraAI/assets/121915052/f5a122a9-b62b-4aa1-b690-96543c4ba492)
try to set the '0.0.0.0' to 'your LAN ip', for me that's fine and works.

**Comment by sizukamasou** [comment]
I did it!
In fact, this is a very simple truth, we all ignore it

Only two tunnels need to be created for service，
The.env file should be configured withnodechatgpt api address.
In addition, since the chat client server is built on the IP port number 3000.
So you only need to add a tunnel to access this port to achieve external network access!
